bench/src/benches/confirmation_slot.rs

```diff
@@ -1,43 +1,107 @@
 use std::path::Path;
+use std::time::Duration;
 
+use crate::metrics::{PingThing, PingThingTxType};
 use crate::tx_size::TxSize;
-use crate::{create_memo_tx, create_rng, send_and_confirm_transactions, Rng8};
-use anyhow::Context;
-use log::{info, warn};
+use crate::{create_memo_tx, create_rng, Rng8};
+use log::{debug, info};
 use solana_rpc_client::nonblocking::rpc_client::RpcClient;
-use solana_sdk::signature::{read_keypair_file, Signer};
+use solana_rpc_client_api::config::{RpcSendTransactionConfig, RpcTransactionConfig};
+use solana_sdk::signature::{read_keypair_file, Signature, Signer};
 use solana_sdk::transaction::Transaction;
 use solana_sdk::{commitment_config::CommitmentConfig, signature::Keypair};
+use solana_transaction_status::UiTransactionEncoding;
+use tokio::time::{sleep, Instant};
+use tracing::error;
 
-/// TC1 send 2 txs (one via LiteRPC, one via Solana RPC) and compare confirmation slot (=slot distance)
+#[derive(Default, PartialEq, Eq)]
+pub struct ConfirmationSlotResult {
+    pub signature: Signature,
+    pub slot_sent: u64,
+    pub slot_landed: u64,
+    pub send_duration: Duration,
+}
+
+#[allow(clippy::too_many_arguments)]
+/// TC1 -- Send 2 txs to separate RPCs and compare confirmation slot.
+/// The benchmark attempts to minimize the effect of real-world distance and synchronize the time that each transaction reaches the RPC.
+/// This is achieved by delaying submission of the transaction to the "nearer" RPC.
+/// Delay time is calculated as half of the difference in duration of [getHealth](https://solana.com/docs/rpc/http/gethealth) calls to both RPCs.
 pub async fn confirmation_slot(
     payer_path: &Path,
     rpc_a_url: String,
     rpc_b_url: String,
     tx_size: TxSize,
+    max_timeout_ms: u64,
+    num_rounds: usize,
+    cu_price_micro_lamports: u64,
+    ping_thing_token: Option<String>,
 ) -> anyhow::Result<()> {
```

Comment:
we need to design a metric result struct - right?

Reply:
going to put this in a different PR to keep things manageable but I was thinking:

CREATE TABLE benchmarks.confirmation_slot (
	signature text NOT NULL,
	bench_datetime timestamptz NOT NULL,
	slot_sent int8 NOT NULL,
	slot_landed int8,
	endpoint text not null,
	confirmed bool not null,
	confirmation_time_ms int4,
	CONSTRAINT signature_slot_pkey PRIMARY KEY (signature, slot)
);